hera_cal/tempcal.py

```diff
@@ -0,0 +1,203 @@
+            # get gain keys
+            keys = list(self.gains.keys())
+            assert isinstance(avg_ants, list)
+            if not isinstance(avg_ants[0], list):
+                avg_ants = [avg_ants]
+            # iterate over antenna lists
+            for antlist in avg_ants:
+                gkeys = [k for k in antlist if k in self.gains]
+                avg = np.sum([self.gains[k] * ~self.gflags[k] for k in gkeys], axis=0) / np.sum([~self.gflags[k] for k in gkeys], axis=0).clip(1e-10, np.inf)
+                avgf = np.min([self.gflags[k] for k in gkeys], axis=0)
+                for gkey in gkeys:
+                    self.gains[gkey] = avg
+                    self.gflags[gkey] = avgf
+
+    def set_abscal_time(self, times, cal_time):
+        """
+        Normalize gains by Julian Date of abscal.
+
+        Args:
+            times : ndarray
+                Times of data in Julian Date.
+            cal_time : float
+                Julian Date of when absolute calibration was performed.
+                This normalizes the gains at this time to one.
+
+        Result:
+            Edits self.gains in place.
+        """
+        tind = np.argmin(np.abs(times - cal_time))
+        for gkey in self.gains:
+            self.gains[gkey] /= abs(self.gains[gkey][tind])
+
+    def gains_from_tempdata(self, tempdata, times, cal_time, temp_coeff):
+        """
+        Convert measurements of temperature data to gains based on temperature
+        coefficient and the time chosen initially for absolute calibration.
+
+        Args:
+            tempdata : DataContainer
+                Holds temperature data [Kelvin] for each auto-correlation
+                as a function of time.
+            times : 1d ndarray
+                Times of data in JD
+            cal_time : float
+                Julian date of initial absolute calibration.
+            temp_coeff : float
+                Temperature coefficient converting dTemp to dGain
+                in units [dGain / dTemp] 
+
+        Result:
+            self.gains : dictionary
+                Holds per-antenna gains given temperature data.
+        """
+        raise NotImplementedError("gains_from_tempdata not yet implemented")
+
+    def write_gains(self, fname, add_to_history='', overwrite=False, verbose=True, **kwargs):
+        """
+        Write self.gains to calfits file.
+
+        Args:
+            fname : str
+                Filepath to output califts file.
+            add_to_history : str
+                Append to output add_to_historystory.
+            overwrite : bool
+                If True overwrite output if it exists
+            verbose : bool
+                Report feedback to stdout
+        
+        Returns:
+            UVCal instance
+                Output calibration object
+        """
+        utils.echo("...writing {}".format(fname), verbose=verbose)
+        uvc = io.write_cal(fname, self.gains, self.freqs, self.times, flags=self.gflags,
```

Comment:
Unless you specifically invoke `return_uvc`, this function doesn't return anything.